sql/000002.sql

```diff
@@ -0,0 +1,183 @@
+-- TODO evaluate perf re partial indexing instead of full unique
+alter table graphile_worker.jobs add column key text unique;
+
+alter table graphile_worker.jobs add locked_at timestamptz;
+alter table graphile_worker.jobs add locked_by text;
```

Comment:
One of the potential advantages this may grant us is the ability to set a concurrency for a particular queue. Currently all queues have concurrency 1. Before merging I'd like to think about this a little deeper (your thoughts are welcome too) because designing the schema in such a way as to easily facilitate this later would be better than schema churn.

Reply:
We should make sure the existing jobs table is updated to honour this, in case there are any jobs in flight. I've not sufficiently validated this SQL, so please check it carefully!

```suggestion
alter table graphile_worker.jobs add locked_by text;

update graphile_worker.jobs
  set locked_at = q.locked_at, locked_by = q.locked_by
  from graphile_worker.job_queues q
  where q.queue_name = jobs.queue_name
  and q.locked_at is not null;
```